File: src/test/java/com/agiledeveloper/WordCount.java
package com.agiledeveloper;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class WordCount {
  public static long count(String path, String searchWord) {
    try {
      final Path resources = Paths.get("src/test/resources");
      final Path p = resources.resolve(path);
      return Files.lines(p)
              .filter(line -> line.contains(searchWord))
              .count();
    } catch(IOException ex) {
      throw new RuntimeException(new FileNotFoundException(ex.getMessage()));
    }
  }
}

File: src/test/scala/com/agiledeveloper/WordCount.scala
package com.agiledeveloper

import scala.io.Source
import java.io._
import java.nio.file.{ Paths, Path }


object WordCount {
  def count(path: String, searchWord: String): Long =
    try {
      val resources = Paths get "src/test/resources"
      val file = resources.resolve(path).toFile

      Source.fromFile(file)
        .getLines
        .filter(_.contains(searchWord))
        .size
    } catch {
      case ex: IOException =>
        throw new RuntimeException(new FileNotFoundException(ex.getMessage))
    }
}
